persistence/src/commonMain/kotlin/com/wire/kalium/persistence/backup/DatabaseExporter.kt

```diff
@@ -36,8 +52,56 @@ internal class DatabaseExporterImpl internal constructor(
     #define SQLITE_CHECKPOINT_RESTART  2  /* Like FULL but wait for readers */
     #define SQLITE_CHECKPOINT_TRUNCATE 3  /* Like RESTART but also truncate WAL */
      */
-    override fun beforeBackup() {
-        sqlDriver.executeQuery(null, "PRAGMA wal_checkpoint(3)", {}, 0)
+//     override fun beforeBackup() {
+//         sqlDriver.executeQuery(null, "PRAGMA wal_checkpoint(3)", {}, 0)
+//     }
+
+    override fun exportToPlainDB(): String? {
+        // delete the backup DB file if it exists
+        if (deleteBackupDBFile()) {
+            return null
+        }
+
+        // create a new backup DB file
+        val plainDatabase: UserDatabaseBuilder =
+            userDatabaseBuilder(platformDatabaseData, backupUserId, null, KaliumDispatcherImpl.io, false)
+
+        // copy the data from the user DB to the backup DB
+        sqlDriver.execute(null, "ATTACH DATABASE ? AS $PLAIN_DB_ALIAS", 1) {
+            bindString(1, plainDatabase.dbFileLocation())
+        }
+
+        try {
+            plainDatabase.database.transaction {
+                dumpContent()
+            }
+        } finally {
+            sqlDriver.execute(null, "DETACH DATABASE $PLAIN_DB_ALIAS", 0)
+        }
+
+        return plainDatabase.dbFileLocation()
```

Comment:
Are we planning to add some tests for the `DatabaseExporter` class?

Reply:
i would do it later first thing tomorrow but today i need to get this change merged